interfaces/builtin/checkbox_support.go

```diff
@@ -0,0 +1,54 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2024 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+const checkboxSupportSummary = `allows checkbox to execute arbitrary system tests`
```

Comment:
I wonder if maybe there's a more generic name we could use here. Checkbox cannot be only tool for executing system wide tests.

Reply:
We agreed to call this checkbox-support as it is, for the moment, only going to be granted to checkbox. I don't mind making it more generic but this reflects the agreement from the engineering sprint.

Reply:
I don't see how the name affects the application. At most you need one line in a snapcraft.yaml file. I don't mind the name being anything specific, to be clear, I'm just going with what was discussed before.

Do we have any specific alternative names?

Reply:
I would hint at the purpose of the interface, something like `system_tester`. My point is that I worry that in X years somebody will have to maintain the snaps and they will get confused no matter how much documentation I spread around that statement if the two names are too similar.

Reply:
@pedronis perhaps we can rename the interface to `system-test-support`? I'm not quite sure what the naming scheme to use here.